### PR TITLE
CLC-6461 Update E-Grades CSV export for Campus Solutions compatibility

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/CLC-6461

### CODE REVIEW

File: app/models/canvas_lti/egrades.rb [hunk 1]
```diff
@@ -19,27 +19,32 @@ def initialize(options = {})
 
     def official_student_grades_csv(term_cd, term_yr, ccn, type)
       raise ArgumentError, 'type argument must be \'final\' or \'current\'' unless GRADE_TYPES.include?(type)
-      official_students = official_student_grades(term_cd, term_yr, ccn)
-      CSV.generate do |csv|
-        csv << %w(student_id grade comment)
-        official_students.each do |student|
-          comment = (student[:pnp_flag] == 'Y') ? 'Opted for P/NP Grade' : ''
-          student_id = student[:student_id]
+      # Campus Solutions expects Windows-style line endings.
```
Comment (johncrossman) [inline app/models/canvas_lti/egrades.rb]: image

Comment (johncrossman) [inline app/models/canvas_lti/egrades.rb]: Campus Solutions is the newest of technologies. Just like Windows 95.

**Comment by johncrossman** [comment]
Looks good, @paulkerschen  I posted one question. Perhaps @raydavis can give a gander, too.

File: app/models/canvas_lti/egrades.rb [hunk 2]
```diff
@@ -49,25 +54,20 @@ def resolve_issues(enable_grading_scheme = false, unmute_assignments = false)
...
-            user_grade = student_grade(course_user['enrollments'])
-            course_students << {
-              :sis_login_id => course_user['sis_login_id'],
-              :final_grade => user_grade[:final_grade],
-              :current_grade => user_grade[:current_grade],
-            }
-          end
+        proxy = Canvas::CourseUsers.new(course_id: @canvas_course_id, paging_callback: self)
+        course_users = proxy.course_users(cache: false)[:body] || []
+        course_users.map do |course_user|
```
Comment (paulkerschen) [inline app/models/canvas_lti/egrades.rb]: @johncrossman Yes, it's a standard proxy call, so will return a well-formed hash of some sort even on error.

**Comment by raydavis** [comment]
Gandered!

**Pull request merged**